runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -1013,6 +1013,16 @@ J9::CodeGenerator::lowerTreeIfNeeded(
       self()->lowerNonhelperCallIfNeeded(node, tt);
       }
 
+   if (node->getOpCode().isCall() &&
+       !node->getSymbol()->castToMethodSymbol()->isHelper() &&
+       (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_invokeBasic))
+      {
+      TR::SymbolReference *vmThreadTempSlotSymRef = self()->comp()->getSymRefTab()->findOrCreateVMThreadTempSlotSymbolRef();
+      TR::Node *numArgsNode = TR::Node::iconst(node, node->getNumArguments() - 1);
+      TR::Node *storeNode = TR::Node::createStore(vmThreadTempSlotSymRef, numArgsNode, TR::istore);
```

Comment:
We should call `createStore(TR::Node *originatingByteCodeNode, TR::SymbolReference * symRef, TR::Node * value)` and pass in the call node as the `originatingByteCodeNode` such that the store node gets the right bytecode index.

Reply:
There is no API for creating store that allows me to specify the opcode as well as the originating bytecode. Without explicitly specifying `TR::istore`, we end up with `TR::astore`. As a workaround for setting the bytecode index, I've added the following line:
```c++
storeNode->setByteCodeIndex(node->getByteCodeIndex());
```